### PR TITLE
Arconomy expansion pack 1: Convenience and Luxury goods

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
This PR adds a few minor additions to the goodie items available through cargo. It's not a terribly long list but these were a few items floating around while I've been handling a few personal things that happened this month.

Items added:
**Mutadone pill:**
We've all been there, for some reason or another, you physically couldn't get to maintenance or shelter in time before a radstorm hits. The sheer quantity of debilitating effects is honestly quite staggering, and your whole body's DNA looks like an Andy Warhol painting. What you need is mutadone, but in the case that a medical department doesn't exist, getting it yourself can be quite the impossible feat. You need to hack into multiple doors and USE multiple pieces of machinery to mix the correct chemicals in the right doses, all just to get a few units of mutadone.
This adds a small, but available alternative to that trial by allowing for mutadone pills to be purchased through cargo. So as to not effect the balance of other, more desirable genetics effects, it's an pill as opposed to a raw syringe making it only really useful for self application or the risky application on others.
It's currently available for 125 credits (All prices subject to change).

**Rapid Lighting Device (RLD):**
The station has blown every possible light. You're at wits end, who wants to go through and replace all these lights? Or maybe you've just built a whole new room, and you're groaning at the prospect of having to go and manually install all of these new lights. If only there was an easier way?
The RLD has technically been a fully functional item for about 4-5 years now, but kept as a debug item for... some reason. I previously added the mini RLD to the illumination BEPIS tech with the feature's inception but it's never really seen that high of a level of use. This adds the full RLD as an otherwise purchasable piece of engineering equipment for the cost of 500 credits.

**Translation Keys:** (Huge thanks to LemonInTheDark for some additional help in debugging).
You're clocking in for your shift in the cargo bay, and you hear the usual chatter of your coworkers. Moffic, Draconic, Calcite, all of them speaking these unusual languages that you're unfamiliar with. If only there was some OTHER WAY to learn what they're saying? Writing messages back and forth between sheets of paper is messy, and lord knows what they meant with all that sign language.
Thankfully, there is! All new Translator keys from CentCom. When worn in a radio headset, temporarily grants the wearer the ability to understand that language until the headset is removed or destroyed. Each of these is a luxury item, so as to not devalue the uniqueness of playing a specific race for specialized communications, currently available for 600 credits each.
image
Additional note: The binary radio key that sounds like it translate the binary machine language now actually does that as part of this content expansion.

## Why It's Good For The Game

Players have expressed repeated interest in adding more content that needs to be obtained via credits to add value to economy as a system. Obviously, this isn't a large enough PR to fully fix that problem, but this adds a few items that are useful and convenient to the player as tools that are gated by some amount of interaction with the economy. This also gave me an opportunity to document several areas of the code as I touched them throughout the process of opening this PR.

The mutadone injector is quite helpful to purchase if a player is disabled via genetics and can't interact with UIs for long periods of times, making the struggle of fixing your genetic state manageable on lower population stations.  That said, I'm not quite sure if I've made this too cheap, however, as mutadone used on certain genetics powersets may be far more subject to balance that I could remember, so I'm open to suggestions on that.

The RLD is not only functional, but also rather convenient. Since the mini RLD is already locked behind an economy feature, moving the parent as a single use item to purchase seems like a decent idea, but may benefit from raising the price to be more in line with the cost of purchasing a tech from the BEPIS. itself. Still, it has a valid gameplay niche and I'd love to see that niche filled in regular gameplay.

Translator keys, honestly, I thought were in the game already. Once I got these working, it seems like the kind of shoe-in content that's not a make-or-break feature to someone's in-round immersion, but if they're running around their department with someone who won't speak common, it might make for a fun feature. 

Footnote more than anything but as I was going through this PR I documented a few areas of the code that were previously left alone, I don't know if I fully touched everything as I went but if anything It didn't make too much sense to atomize for purely code documentation in my mind.

**I would ask that these items not be moved to lathes or maps for the foreseeable future.** The point is that this is meant to bolster the number of items added to cargo for the crew to purchase, not to add these items for usage sake.

## Changelog

:cl:
add: Added several new goodie items to purchase through cargo, including translation keys, mutadone autoinjectors, and a full RLD.
code: documented research_nodes and radios.
/:cl:

### CODE REVIEW

**Comment by Pandarsenic** [comment]
Some of these things are actually very cool; I'd be interested in seeing an expansion of things that don't come through cargo, though, because cargo can be unreliable to have at all. Adding mutadone medipens to some sort of on-station vendors that aren't linked to cargo seems like it might be really cool?

**Comment by YuliceTheDruid** [comment]
Medipens are actually more useful than raw syringes because they have instant injection both on the user AND anyone they target. Making a medipen Mutadone injector will basically let anyone take it and run up to someone in melee range and instantly inject them.

Probably going to hate me for bringing it up, but this would basically kill the usefulness of hulk as every single sec officer is now going to start carrying one so they can just jab a hulk before batonning them. Might be wiser to make it a pill or a patch if you're worried about easy syringe gun ammo for it.

**Comment by optimumtact** [comment]
yes the form it's delivered in is wrong

**Comment by moocowswag** [comment]
> Probably going to hate me for bringing it up, but this would basically kill the usefulness of hulk as every single sec officer is now going to start carrying one so they can just jab a hulk before batonning them. 

WIth the time it takes to order and get something from cargo a hulk could have already rampaged several departments. Its a very specific counter item for a very specific department.

Lets take this into the context of revs for example.

And lets give a perfect example of when mutadone auto injectors would come into play, revs took over sci, medbay is contested, and cargo is under sec`s rule (which is normally what sec first take over during revs)

With mutadone auto injectors sec can inject isolated hulks and baton + mindshield them..sounds really good on paper but if medbay is a contested department and sci is taken over by revs then its more than likely that revs have naturally gotten their numbers advantage and so you won`t get a situation where you can autoinject an isolated hulk unless they are braindead.

As for non rev rounds then it remains a specific counter for a specific enemy.


Now considering that it will take a few minutes to get a mutadone auto injector to respond to a bad hulk, what about rushing for them before you ever hear about the existence of a hulk.

Metagame aside this isnt entirely viable either because there are several other likely niche counters for niche department bad guys, emp for augmented people, mutadone pills for chemist with a syringe gun, flash for borgs, laser guns for simple mob spam from xenobio, modsuit for atmos tech / ai plasmaflooding, tools/insulation to hack with for malf ai bolting and shocking doors.


The list for specific counters can go on and its very unlikely sec are gonna spend time buying mutadone auto injectors for hulk when they have to otherwise gather gear to counter so many other things, especially after the toolbelt nerf made it so they cant storage every tool in their backpack.

**Comment by Palpetinus** [comment]
> WIth the time it takes to order and get something from cargo a hulk could have already rampaged several departments. Its a very specific counter item for a very specific department.

Playing both sec and medical a lot, hulks are a very common sight lately. Secoffs (or, I would) will just round start head to cargo as part of the "shift start routine", order a pen and go on with the shift. Privately purchased goods can only be opened by the buyer anyway, so there's not even a risk of cargonians stealing it.
Additionally, it pretty much removes one of the very, very few reasons for the syringe gun to exist.

**Comment by YuliceTheDruid** [comment]
> > Probably going to hate me for bringing it up, but this would basically kill the usefulness of hulk as every single sec officer is now going to start carrying one so they can just jab a hulk before batonning them.
> 
> WIth the time it takes to order and get something from cargo a hulk could have already rampaged several departments. Its a very specific counter item for a very specific department.
> 
> Lets take this into the context of revs for example.
> 
> And lets give a perfect example of when mutadone auto injectors would come into play, revs took over sci, medbay is contested, and cargo is under sec`s rule (which is normally what sec first take over during revs)
> 
> With mutadone auto injectors sec can inject isolated hulks and baton + mindshield them..sounds really good on paper but if medbay is a contested department and sci is taken over by revs then its more than likely that revs have naturally gotten their numbers advantage and so you won`t get a situation where you can autoinject and baton an isolated hulk unless they are braindead, and then they can just go back to genetics and get hulk back while you get swarmed by 3 assistants.
> 
> As for non rev rounds then it remains a specific counter for a specific enemy.
> 
> Now considering that it will take a few minutes to get a mutadone auto injector to respond to a bad hulk, what about rushing for them before you ever hear about the existence of a hulk.
> 
> Metagame aside this isnt entirely viable either because there are several other likely niche counters for niche department bad guys, emp for augmented people, mutadone pills for chemist with a syringe gun, flash for borgs, laser guns for simple mob spam from xenobio, modsuit for atmos tech / ai plasmaflooding, tools/insulation to hack with for malf ai bolting and shocking doors.
> 
> The list for specific counters can go on and its very unlikely sec are gonna spend time buying mutadone auto injectors for hulk when they have to otherwise gather gear to counter so many other things, especially after the toolbelt nerf made it so they cant store every tool in their backpack without taking up a ton of room.

Sure, there's other shit that they could get counters for, but hulk remains one of the few things that lets people ignore batons, and batons are far and away the best piece of equipment sec has because we want to encourage them to nonlethally detain people. Hell, I'd argue that admins would let buying a bunch roundstart as sec slide since most people hate hulk and it'd encourage nonlethally detaining instead of killing people. Augmented people are a joke and still die to a baton, as does a chemist and they can be countered by just wearing a modsuit/thick clothing or not getting hit, and again, also instanyl get fucked by a baton. Flashes are stock carry for sec officers anyway so I dunno why you'd bring that up, and simple mob spam is a lot harder/longer to achieve/set up than just grabbing a hulk syringe and going ham. as for pflooding, I've not seen a genuine pflood in my entire time playing here but that could just be a consequence of me playing Manuel.

**Comment by JoshAdamPowell** [comment]
FINALLY the RLD will be purchasable! I never understood why it was admin-only

**Comment by Pandarsenic** [comment]
> Hell, I'd argue that admins would let buying a bunch roundstart as sec slide since most people hate hulk and it'd encourage nonlethally detaining instead of killing people. 

This sounds good to me because hulktide is annoying as hell.

Also,

> **An** mutadone medipen

**Comment by anclandor** [comment]
Can you make it so translation key doesn't translate codespeak?  For one, if traitors/ops spend the tc just so they can have the peace of mind of not being understood, they should be able to have that peace of mind, and for two, it just feels kind of lame if NT can just crack the syndicates fancy codespeak with a little headset gadget.  (note, if you did this already than sorry for being lazy and not looking at the code and kudos to you)

**Comment by ArcaneMusic** [comment]
The Autoinjector has been changed to a pill to better avoid issues with usage time.

**Comment by Striders13** [comment]
RLD like completely 100% counters the nightmare, anything can be done about that? Because when maint is filled with a light on every tile that needs to be clicked you are really limited in your movement.

File: code/game/objects/items/devices/radio/headset.dm [hunk 1]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 		return attack_self(headset_user)
 	return ..()
 
+/obj/item/radio/headset/proc/learn_language()
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:88]: SIGNAL_HANDLER

File: code/game/objects/items/devices/radio/headset.dm [hunk 2]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+	if(keyslot?.translated_language)
+		language_list += keyslot.translated_language
+	if(keyslot2?.translated_language)
+		language_list += keyslot2.translated_language
+
+	var/mob/living/carbon/person = loc
+	for(var/iterator in language_list)
+		person.grant_language(iterator, TRUE, FALSE, LANGUAGE_RADIOKEY)
+
+/obj/item/radio/headset/proc/unlearn_language()
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:101]: SIGNAL_HANDLER

File: code/game/objects/items/devices/radio/headset.dm [hunk 3]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+		return
+	var/list/language_list = list()
+	if(keyslot?.translated_language)
+		language_list += keyslot.translated_language
+	if(keyslot2?.translated_language)
+		language_list += keyslot2.translated_language
+
+	var/mob/living/carbon/person = loc
+	for(var/iterator in language_list)
+		person.grant_language(iterator, TRUE, FALSE, LANGUAGE_RADIOKEY)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:99]: Named parameters (TRUE and FALSE)

File: code/game/objects/items/devices/radio/headset.dm [hunk 4]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+		return
+	var/list/language_list = list()
+	if(keyslot?.translated_language)
+		language_list += keyslot.translated_language
+	if(keyslot2?.translated_language)
+		language_list += keyslot2.translated_language
+
+	var/mob/living/carbon/person = loc
+	for(var/iterator in language_list)
+		person.remove_language(iterator, TRUE, FALSE, LANGUAGE_RADIOKEY)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:112]: Named parameters (TRUE and FALSE)

File: code/game/objects/items/devices/radio/headset.dm [hunk 5]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+	if(!istype(loc, /mob/living/carbon))
+		return
+	var/list/language_list = list()
+	if(keyslot?.translated_language)
+		language_list += keyslot.translated_language
+	if(keyslot2?.translated_language)
+		language_list += keyslot2.translated_language
+
+	var/mob/living/carbon/person = loc
+	for(var/iterator in language_list)
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:98]: Better variable name, just use "language"

File: code/game/objects/items/devices/radio/headset.dm [hunk 6]
```diff
@@ -83,6 +85,32 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
```
Comment (Mothblocks) [inline code/game/objects/items/devices/radio/headset.dm:111]: Better variable name, just use "language"

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**